### PR TITLE
Fix ContinuousBernoulli numerical stability

### PR DESCRIPTION
**Pull request by vsimkus**
Fixes a typo in the Taylor approximation of the log-normalising constant https://github.com/tensorflow/probability/issues/1511, which yielded negative KL values.

Also fixes the stability of the gradients of the mean (and thus the KL). The existing implementation was not stable for large negative logits:
```python
import numpy as np
import tensorflow_probability as tfp
logits = np.linspace(-100, 100, 20)[..., np.newaxis].astype(np.float32)
_, grad_logits = tfp.math.value_and_gradient(lambda x: tfp.distributions.ContinuousBernoulli(logits=x).mean(), logits)
grad_logits
# <tf.Tensor: shape=(20, 1), dtype=float32, numpy=array([[          nan], [          nan], [1.6044446e-04], [2.1360947e-04], [2.9834712e-04], [4.4567912e-04], [7.3673454e-04], [1.4439999e-03], [4.0109721e-03], [3.0867012e-02], [3.0867014e-02], [4.0109721e-03], [1.4439999e-03], [7.3673454e-04], [4.4567912e-04], [2.9834712e-04], [2.1360947e-04], [1.6044446e-04], [1.2491349e-04], [9.9999997e-05]], dtype=float32)>
```

And here's the same output from the new implementation:
```python
import numpy as np
import tensorflow_probability as tfp
logits = np.linspace(-100, 100, 20)[..., np.newaxis].astype(np.float32)
_, grad_logits = tfp.math.value_and_gradient(lambda x: tfp.distributions.ContinuousBernoulli(logits=x).mean(), logits)
grad_logits
# <tf.Tensor: shape=(20, 1), dtype=float32, numpy=array([[9.9999997e-05], [1.2491349e-04], [1.6044446e-04], [2.1360947e-04], [2.9834712e-04], [4.4567912e-04], [7.3673454e-04], [1.4439999e-03], [4.0109721e-03], [3.0867012e-02], [3.0867014e-02], [4.0109721e-03], [1.4439999e-03], [7.3673454e-04], [4.4567912e-04], [2.9834712e-04], [2.1360947e-04], [1.6044446e-04], [1.2491349e-04], [9.9999997e-05]], dtype=float32)>
```

### CODE REVIEW

**Comment by vsimkus** [comment]
You're welcome, I'm glad it is accepted now.
